app/controllers/questions_controller.rb

```diff
@@ -18,9 +21,34 @@ def index
     @questions = Question.all
   end
 
+  def show
+    @question = Question.find(params[:id])
+    @choices = @question.choices
+  end
+
+  def edit
+    @question = Question.find(params[:id])
+    @choices = @question.choices
+  end
+
+  def update
+    @question = Question.find(params[:id])
+    if @question.update_attributes(question_params)
+      flash[:notice] = 'Question updated.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.